skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkatePluginSettings.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2023 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.sgp.intellij
+
+import com.intellij.openapi.components.BaseState
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.SimplePersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import com.intellij.openapi.project.Project
+
+@Service(Service.Level.PROJECT)
+@State(name = "SkatePluginSettings", storages = [Storage("skate.xml")])
+class SkatePluginSettings(
```

Comment:
Let's add some description about each class

Reply:
Is a description like "manages user-specific settings for the skate plugin" sufficient?

Reply:
As an example:

https://softwareengineering.stackexchange.com/questions/187882/how-can-i-become-better-on-explaining-the-code-to-other-developers

https://towardsdatascience.com/5-tips-to-write-better-comment-in-your-code-1d4681071967

As you are adding more code, I want you to think about readability of the code. Is it easier for someone else to read your code and makes sense of your action and intention? How your documentation can help etc.

Reply:
dd48f90